alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt

```diff
@@ -867,4 +868,228 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
+        indexDoc(testSourceIndex, "1", docPayload.toString())
+        // Create monitor
+        try {
+            createMonitor(monitor)
+        } catch (e: Exception) {
+            assertTrue(e.message?.contains("can't process index [$testSourceIndex] due to field mapping limit") ?: false)
+        }
+    }
+
+    fun `test queryIndex not rolling over multiple monitors`() {
+        val testSourceIndex = "test_source_index"
+        createIndex(testSourceIndex, Settings.EMPTY)
+
+        val docQuery = DocLevelQuery(query = "test_field:\"us-west-2\"", name = "3")
+        val docLevelInput = DocLevelMonitorInput("description", listOf(testSourceIndex), listOf(docQuery))
+        val trigger = randomDocumentLevelTrigger(condition = ALWAYS_RUN)
+        var monitor = randomDocumentLevelMonitor(
+            inputs = listOf(docLevelInput),
+            triggers = listOf(trigger)
+        )
+        // This doc should create close to 1000 (limit) fields in index mapping. It's easier to add mappings like this then via api
+        val docPayload: StringBuilder = StringBuilder(100000)
+        docPayload.append("{")
+        for (i in 1..10) {
+            docPayload.append(""" "id$i":$i,""")
+        }
+        docPayload.append("\"test_field\" : \"us-west-2\" }")
+        indexDoc(testSourceIndex, "1", docPayload.toString())
+        // Create monitor #1
+        var monitorResponse = createMonitor(monitor)
+        assertFalse(monitorResponse?.id.isNullOrEmpty())
+        monitor = monitorResponse!!.monitor
+        // Execute monitor #1
+        var executeMonitorResponse = executeMonitor(monitor, monitorResponse.id, false)
+        Assert.assertEquals(executeMonitorResponse!!.monitorRunResult.monitorName, monitor.name)
+        Assert.assertEquals(executeMonitorResponse.monitorRunResult.triggerResults.size, 1)
+        // Create monitor #2
+        var monitorResponse2 = createMonitor(monitor)
+        assertFalse(monitorResponse2?.id.isNullOrEmpty())
+        monitor = monitorResponse2!!.monitor
+        // Insert doc #2. This one should trigger creation of alerts during monitor exec
+        val testDoc = """{
+            "test_field" : "us-west-2"
+        }"""
+        indexDoc(testSourceIndex, "2", testDoc)
+        // Execute monitor #2
+        var executeMonitorResponse2 = executeMonitor(monitor, monitorResponse2.id, false)
+        Assert.assertEquals(executeMonitorResponse2!!.monitorRunResult.monitorName, monitor.name)
+        Assert.assertEquals(executeMonitorResponse2.monitorRunResult.triggerResults.size, 1)
+
+        refreshIndex(AlertIndices.ALERT_INDEX)
+        var alerts = searchAlerts(monitorResponse2.id)
+        Assert.assertTrue(alerts != null)
+        Assert.assertTrue(alerts.size == 1)
+
+        // Both monitors used same queryIndex. Since source index has close to limit amount of fields in mappings,
+        // we expect that creation of second monitor would rollover queryIndex
```

Comment:
Wait is the expectation that the `queryIndex` is rolled over or not? It is not getting rolled over here.

Reply:
This comment was incorrect. I fixed it with new commit and added few more in other tests